src/java/edu/ucsd/library/jollyroger/JollyRoger.java

```diff
@@ -109,7 +109,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
 		String roger = null;
 		if ( type.equals("bib") )
 		{
-			if ( value.length() > 8 ) { value = value.substring(0,8); }
+			if (value.indexOf("~") > 0) { value = value.substring(0, value.indexOf("~")).trim(); }
```

Comment:
This seems good. Honestly I thought bibs had to be 8 digits. I wonder if there's more granular info on that we could use for parsing. But, :+1: on this solution.